STM32/Util/Src/faultHandlers.c

```diff
@@ -0,0 +1,110 @@
+/** 
+ ******************************************************************************
+ * @file    faultHandlers.c
+ * @brief   This file contains fault handlers and crash info
+ * @date:   19 APR 2024
+ * @author: Luke W
+ ******************************************************************************
+*/
+
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include "stm32f4xx_hal.h"
+
+#include "faultHandlers.h"
+#include "FLASH_readwrite.h"
+#include "USBprint.h"
+
+/***************************************************************************************************
+** DEFINES
+***************************************************************************************************/
+
+#define BUF_LEN 400
+
+/***************************************************************************************************
+** TYPEDEFS
+***************************************************************************************************/
+
+
+
+/***************************************************************************************************
+** PRIVATE VARIABLES
+***************************************************************************************************/
+
+/* Defined in the linker script */
+extern uint32_t _FlashAddr;
```

Comment:
Is it the intention that all boards incorporate this strategy as fault handlers on all boards? If so, then I foresee a bit of a FLASH sector problem as this would be the 3rd out of 8 sectors reserved for board data and filled mostly with 0s.  

The way this is implemented, using _FlashAddr, the faultHandler module reserves a FLASH sector for itself or at least removes any information that was in that sector previously. Not a problem for the AC board ofc, but generally we might run into issues.

I think we might need to spend some time thinking about how the relatively large FLASH sectors can be used more effectively.